### PR TITLE
add metros to explore location search

### PR DESCRIPTION
**Pull request by pnavarrc**
- Centralized the logic with the global location search

### CODE REVIEW

File: src/components/Explore/utils.ts [hunk 1]
```diff
@@ -429,42 +420,9 @@ export function getSubtitle(
...
-        ...sortedStateCounties,
-        ...sortedOtherCounties,
-      ]
-    : [
-        ...regions.customAreas,
-        ...sortedStateCounties,
-        ...sortedStates,
-        ...sortedOtherCounties,
-      ];
+  return currentLocation ? getSearchAutocompleteLocations(currentLocation) : [];
```
Comment (ghop02) [inline src/components/Explore/utils.ts:425]: This feels a bit like it's tying two different areas together - thought on making a generic `getAutoCompleteLocations(region?: Region)` that both use?

Comment (pnavarrc) [reply src/components/Explore/utils.ts:425]: Yeah, I can move the function to a more central place, as the logic is the same for both

File: src/common/regions/utils.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export function belongsToState(county: County, stateFips: string | null) {
+  return county.state.fipsCode === stateFips;
+}
+
+export function getAutocompleteLocations(region?: Region) {
+  const { states, metroAreas, counties } = regions;
+
+  // Homepage
+  if (!region) {
+    return concat<Region>(states, metroAreas, counties);
```
Comment (ghop02) [inline src/common/regions/utils.ts:15]: we want to include USA and NAMC here right?

File: src/common/regions/utils.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    const [countiesInMetro, otherCounties] = partition(counties, county =>
+      region.counties.includes(county),
+    );
+    return concat<Region>(countiesInMetro, states, otherCounties, metroAreas);
+  } else if (region instanceof State || region instanceof County) {
+    const stateFips = getStateFips(region);
+    const [stateCounties, otherCounties] = partition(counties, county =>
+      belongsToState(county, stateFips),
+    );
+    return concat<Region>(stateCounties, states, metroAreas, otherCounties);
```
Comment (ghop02) [inline src/common/regions/utils.ts:29]: Happy to fix this up in a separate PR, but I would love to partition the metros to pull out metros that have counties in the state

File: src/common/regions/utils.ts (hunk 2 above)
Comment (pnavarrc) [reply src/common/regions/utils.ts:15]: No, they don't have location pages - I'm adding them back on the autocomplete for explore

File: src/common/regions/utils.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { concat, partition } from 'lodash';
+import regions from './region_db';
+import { getStateFips } from './regions_data';
+import { County, State, Region, MetroArea } from './types';
+
+export function belongsToState(county: County, stateFips: string | null) {
+  return county.state.fipsCode === stateFips;
+}
+
+export function getAutocompleteLocations(region?: Region) {
```
Comment (ghop02) [inline src/common/regions/utils.ts:10]: nit: Would prefer this to be `getAutocompleteRegions` as we have the `Location` and `Region` object. Also, will type inference work properly here? 

Also, maybe a name like `getAllAutocompleteRegions` or `getAllAutocompleteRegionsSorted` might be more clear? Though I don't love those names usggestions.  I think a comment here might be useful to indicate what the `region` argument does.  it's not clear from the function signature that region changes the sort order of the list.

**Review by ghop02 — APPROVED** [review]
whoops meant to approve last time

**Review by chasulin — APPROVED** [review]
Pending the comment re: updating `getFilterLimit` in slack, looks good

**Pull request merged**